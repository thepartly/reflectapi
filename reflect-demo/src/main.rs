#[cfg(test)]
mod tests;

#[derive(reflect::Input)]
struct TestStructWithVec<T>
where
    T: reflect::Input,
{
    _f: T,
}

#[derive(reflect::Input)]
struct TestStructParent
// where
//     T: reflect::Input,
{
    _f: TestStructWithVec<u8>,
    // _f2: TestStructWithVec<T>,
}

// #[derive(serde::Deserialize)]
// struct Test<'a> {
//     _a: std::slice::Iter<'a, u8>,
// }

trait MyTrait {}

fn main() {
    // println!("{}", MyStruct::reflect_input());
    println!("{:#?}", TestStructParent::reflect_input());
    // println!("{:#?}", TestStructWithVec::<u8>::reflect_input());
}
