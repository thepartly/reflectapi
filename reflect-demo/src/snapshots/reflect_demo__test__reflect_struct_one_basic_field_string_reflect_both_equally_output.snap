---
source: reflect-demo/src/main.rs
expression: "TestStructOneBasicFieldStringReflectBothEqually::reflect_output()"
---
{
  "name": "",
  "types": [
    {
      "name": "reflect_demo::test::TestStructOneBasicFieldStringReflectBothEqually",
      "fields": [
        {
          "name": "_f",
          "type": {
            "name": "u32"
          }
        }
      ]
    },
    {
      "name": "u32"
    }
  ]
}
